File: 21.07.2024/Longest Substring Without Repeating Characters.c
int lengthOfLongestSubstring(char * s) {
    int map[128] = {0}, max_len = 0, start = 0;
    for (int end = 0; s[end]; end++) {
        start = fmax(start, map[s[end]]);
        map[s[end]] = end + 1;
        max_len = fmax(max_len, end - start + 1);
    }
    return max_len;
}

File: 20.07.2024/Longest Substring Without Repeating Characters.java
public class LongestSubstring {
    public static int lengthOfLongestSubstring(String s) {
        int max = 0;
        Map<Character, Integer> map = new HashMap<>();
        for (int start = 0, end = 0; end < s.length(); end++) {
            if (map.containsKey(s.charAt(end))) {
                start = Math.max(map.get(s.charAt(end)) + 1, start);
            }
            map.put(s.charAt(end), end);
            max = Math.max(max, end - start + 1);
        }
        return max;
    }
}
